crates/sui-storage/src/lock_service.rs

```diff
@@ -83,8 +83,65 @@ enum LockServiceQueries {
     },
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub enum ObjectLockStatus {
+    RequestedObjectRefLocked {
```

Comment:
This seems a bit confusing because if locked_by_tx is None, it's actually not locked?

Reply:
I had a hard time coming up with good names with this. Any suggestions?

Reply:
I think it's hard probably because you are combining two cases in one enum. Would it be too complex if we break this down to two cases?

Reply:
I wanted  to convince myself that "locked" and "locked by a tx" is different (we indeed use these terms interchangeably in the code base). If we want very clear distinction, we have to make it more verbose, e.g. `RequestedObjectRefLockedOrInitialized` where `initialized` means the row is there but not yet "locked" by any tx

Reply:
Agreed these are indeed confusing. Since the top level enum name already has Object in it, I think the kinds can be more concise.
What about the following 3 enums:
- Initialized
- LockedToTx
- LockedAtDifferentVersion

Reply:
updated @lxfind